allensdk/api/cloud_cache/cloud_cache.py

```diff
@@ -0,0 +1,305 @@
+                subset = self.s3_client.list_objects_v2(Bucket=self._bucket_name,  # noqa: E501
+                                                        Prefix='manifests/',
+                                                        ContinuationToken=continuation_token)  # noqa: E501
+            else:
+                subset = self.s3_client.list_objects_v2(Bucket=self._bucket_name,  # noqa: E501
+                                                        Prefix='manifests/')
+
+            if 'Contents' in subset:
+                for obj in subset['Contents']:
+                    output.append(os.path.basename(obj['Key']))
+
+            if 'NextContinuationToken' in subset:
+                continuation_token = subset['NextContinuationToken']
+            else:
+                keep_going = False
+
+        output.sort()
+        return output
+
+    def load_manifest(self, manifest_name: str):
+        """
+        Load a manifest from this dataset.
+
+        Parameters
+        ----------
+        manifest_name: str
+            The name of the manifest to load. Must be an element in
+            self.manifest_file_names
+        """
+        if manifest_name not in self.manifest_file_names:
+            raise ValueError(f"manifest: {manifest_name}\n"
+                             "is not one of the valid manifest names "
+                             "for this dataset:\n"
+                             f"{self.manifest_file_names}")
+
+        manifest_key = 'manifests/' + manifest_name
+        stream = io.BytesIO()
+        response = self.s3_client.get_object(Bucket=self._bucket_name,
+                                             Key=manifest_key)
+        for chunk in response['Body'].iter_chunks():
+            stream.write(chunk)
+        stream.seek(0)
+        self._manifest.load(stream)
+
+    def _file_exists(self, file_attributes: CacheFileAttributes) -> bool:
+        """
+        Given a CacheFileAttributes describing a file, assess whether or
+        not that file exists locally and is valid (i.e. has the expected
+        file hash)
+
+        Parameters
+        ----------
+        file_attributes: CacheFileAttributes
+            Description of the file to look for
+
+        Returns
+        -------
+        bool
+            True if the file exists and is valid; False otherwise
+
+        Raises
+        -----
+        RuntimeError
+            If file_attributes.local_path exists but is not a file.
+            It would be unclear how the cache should proceed in this case.
+        """
+
+        if not os.path.exists(file_attributes.local_path):
+            return False
+        if not os.path.isfile(file_attributes.local_path):
+            raise RuntimeError(f"{file_attributes.local_path}\n"
+                               "exists, but is not a file;\n"
+                               "unsure how to proceed")
+
+        full_path = str(file_attributes.local_path.resolve())
```

Comment:
and since `file_path_from_hash` just `open`s this object, it can be a `Path` so you don't need to make it a string.

Reply:
done